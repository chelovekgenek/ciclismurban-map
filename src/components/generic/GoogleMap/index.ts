import { Marker } from "react-google-maps"
import { default as GMap } from "./GoogleMap"

export const GoogleMap = Object.assign(GMap, {
  Marker,
})
