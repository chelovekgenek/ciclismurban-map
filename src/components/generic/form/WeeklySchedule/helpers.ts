import moment, { Moment } from "moment"

import options from "./options.json"

export const parse = (v: string) => moment(v, options.format)
export const format = (v: Moment) => v.format(options.format)
