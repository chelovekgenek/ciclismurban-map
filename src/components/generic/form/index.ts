export * from "./Input"
export * from "./Textarea"
export * from "./Upload"
export * from "./GoogleMap"
export * from "./DateTimePicker"
