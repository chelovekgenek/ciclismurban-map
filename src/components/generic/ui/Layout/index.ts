import { Layout as AntdLayout } from "antd"
import styled from "styled-components"

const Sider = styled(AntdLayout.Sider)`
  &.ant-layout-sider-light {
    background-color: #f5f5f5;
  }
  .ant-layout-sider-children {
    padding: 1rem;
  }
`

export const Layout = Object.assign(AntdLayout, {
  Content: AntdLayout.Content,
  Sider: Sider,
})
