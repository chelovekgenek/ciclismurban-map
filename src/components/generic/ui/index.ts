export * from "./Grid"
export * from "./Empty"
export * from "./Checkbox"
export * from "./GoogleMap"
export * from "./Spinner"
export * from "./Input"
export * from "./Button"
export * from "./Icon"
