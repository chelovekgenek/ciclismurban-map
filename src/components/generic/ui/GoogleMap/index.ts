import { InfoWindow } from "react-google-maps"
import { default as Map } from "./GoogleMap"
import { Marker } from "./Marker"

export const GoogleMap = Object.assign(Map, {
  Marker,
  InfoWindow,
})
