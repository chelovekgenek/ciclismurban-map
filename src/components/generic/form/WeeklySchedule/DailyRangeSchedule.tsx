import React, { useMemo, useCallback } from "react"
import { TimePicker } from "antd"
import { Moment } from "moment"

import { Grid } from "components/generic/layout"
import { Button } from "components/generic/ui"
import { WeeklyScheduleModel } from "models/location"

import * as Styles from "./DailyRangeSchedule.styled"
import options from "./options.json"
import { parse } from "./helpers"

interface IProps {
  value?: WeeklyScheduleModel[keyof WeeklyScheduleModel][number]
  onAdd?: () => void
  onCancel?: () => void
  onChange?: (value: Required<IProps>["value"]) => void
}

export const DailyRangeSchedule: React.FC<IProps> = ({ value, onAdd, onCancel, onChange }) => {
  const from = useMemo(() => (value ? parse(value.from) : undefined), [value])
  const to = useMemo(() => (value ? parse(value.to) : undefined), [value])
  const handleFromChange = useCallback(
    (_time: Moment, timeStr: string) =>
      onChange &&
      onChange({
        from: timeStr,
        to: value!.to,
      }),
    [value, onChange],
  )
  const handleToChange = useCallback(
    (_time: Moment, timeStr: string) =>
      onChange &&
      onChange({
        from: value!.from,
        to: timeStr,
      }),
    [value, onChange],
  )
  return (
    <Grid.Row>
      <Grid.Col span={9}>
        {from ? (
          <TimePicker format={options.format} value={from} onChange={handleFromChange} />
        ) : (
          <Styles.EmptyBox>-</Styles.EmptyBox>
        )}
      </Grid.Col>
      <Grid.Col span={9}>
        {to ? (
          <TimePicker format={options.format} value={to} onChange={handleToChange} />
        ) : (
          <Styles.EmptyBox>-</Styles.EmptyBox>
        )}
      </Grid.Col>
      <Grid.Col span={6}>
        <Styles.ActionWrapper>
          <Button icon="minus" onClick={onCancel} disabled={!onCancel} />
          <Button icon="plus" onClick={onAdd} disabled={!onAdd} />
        </Styles.ActionWrapper>
      </Grid.Col>
    </Grid.Row>
  )
}
