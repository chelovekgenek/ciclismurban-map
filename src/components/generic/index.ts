import * as Layout from "./layout"
import * as UI from "./ui"

export { Layout, UI }
