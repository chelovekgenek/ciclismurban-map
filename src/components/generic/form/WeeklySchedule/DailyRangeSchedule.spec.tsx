import React from "react"
import moment, { Moment } from "moment"
import { shallow } from "enzyme"
import { TimePicker } from "antd"

import { Button } from "components/generic/ui"

import { DailyRangeSchedule } from "./DailyRangeSchedule"
import * as helpers from "./helpers"

describe("pages/DailyRangeSchedule", () => {
  const defaultValues = {
    from: moment("2019-10-29T12:00:00.000Z"),
    to: moment("2019-10-29T20:00:00.000Z"),
  }
  const defaultProps = {
    value: { from: helpers.format(defaultValues.from), to: helpers.format(defaultValues.to) },
  }
  const render = (props: Partial<React.ComponentProps<typeof DailyRangeSchedule>> = {}) =>
    shallow(<DailyRangeSchedule {...defaultProps} {...props} />)

  const getButtonByIcon = (wrapperValues: Partial<React.ComponentProps<typeof DailyRangeSchedule>>, icon: string) => {
    const wrapper = render(wrapperValues)
    return wrapper.find(Button).filterWhere(w => w.props().icon === icon)
  }
  const getTimepickerByValue = (
    wrapperValues: Partial<React.ComponentProps<typeof DailyRangeSchedule>>,
    value: Moment,
  ) => {
    const wrapper = render(wrapperValues)
    return wrapper.find(TimePicker).findWhere(w => moment(w.prop("value")).isSame(value))
  }

  const spy = jest.spyOn(helpers, "parse")
  beforeAll(() => {
    spy.mockImplementation(v => {
      if (v === defaultProps.value.from) {
        return defaultValues.from
      }
      if (v === defaultProps.value.to) {
        return defaultValues.to
      }
      return moment()
    })
  })
  afterAll(() => {
    spy.mockRestore()
  })
  it("should match snapshot if value was not passed", () => {
    expect(render({ value: undefined })).toMatchSnapshot()
  })
  it("should match snapshot if value was passed", () => {
    expect(render()).toMatchSnapshot()
  })
  it("should set minus button enabled if onCancel handler is passed", () => {
    const button = getButtonByIcon({ onCancel: jest.fn() as any }, "minus")
    expect(button.prop("disabled")).toBeFalsy()
  })
  it("should set plus button enabled if onCancel handler is passed", () => {
    const button = getButtonByIcon({ onAdd: jest.fn() as any }, "plus")
    expect(button.prop("disabled")).toBeFalsy()
  })
  it("should call onCancel handler", () => {
    const props = { onCancel: jest.fn() as any }
    const button = getButtonByIcon(props, "minus")
    button.simulate("click")
    expect(props.onCancel).toBeCalledTimes(1)
  })
  it("should call onAdd handler", () => {
    const props = { onAdd: jest.fn() as any }
    const button = getButtonByIcon(props, "plus")
    button.simulate("click")
    expect(props.onAdd).toBeCalledTimes(1)
  })
  it('should call onChange at "from" input', () => {
    const props = { onChange: jest.fn() as any }
    const timepicker = getTimepickerByValue(props, defaultValues.from)
    timepicker.simulate("change", defaultValues.to, defaultProps.value.to)
    expect(props.onChange).toBeCalledWith({
      from: "20:00",
      to: "20:00",
    })
  })
  it('should call onChange at "to" input', () => {
    const props = { onChange: jest.fn() as any }
    const timepicker = getTimepickerByValue(props, defaultValues.to)
    timepicker.simulate("change", defaultValues.from, defaultProps.value.from)
    expect(props.onChange).toBeCalledWith({
      from: "12:00",
      to: "12:00",
    })
  })
})
