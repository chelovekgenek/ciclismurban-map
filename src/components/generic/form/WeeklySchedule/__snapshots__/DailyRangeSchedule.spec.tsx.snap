// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/DailyRangeSchedule should match snapshot if value was not passed 1`] = `
<Row
  gutter={0}
>
  <Col
    span={9}
  >
    <styled.div>
      -
    </styled.div>
  </Col>
  <Col
    span={9}
  >
    <styled.div>
      -
    </styled.div>
  </Col>
  <Col
    span={6}
  >
    <Styled(ButtonGroup)>
      <Button
        block={false}
        disabled={true}
        ghost={false}
        htmlType="button"
        icon="minus"
        loading={false}
      />
      <Button
        block={false}
        disabled={true}
        ghost={false}
        htmlType="button"
        icon="plus"
        loading={false}
      />
    </Styled(ButtonGroup)>
  </Col>
</Row>
`;

exports[`pages/DailyRangeSchedule should match snapshot if value was passed 1`] = `
<Row
  gutter={0}
>
  <Col
    span={9}
  >
    <TimePicker
      align={
        Object {
          "offset": Array [
            0,
            -2,
          ],
        }
      }
      disabled={false}
      focusOnOpen={true}
      format="HH:mm"
      hideDisabledOptions={false}
      onChange={[Function]}
      placement="bottomLeft"
      transitionName="slide-up"
      value={"2019-10-29T11:00:00.000Z"}
    />
  </Col>
  <Col
    span={9}
  >
    <TimePicker
      align={
        Object {
          "offset": Array [
            0,
            -2,
          ],
        }
      }
      disabled={false}
      focusOnOpen={true}
      format="HH:mm"
      hideDisabledOptions={false}
      onChange={[Function]}
      placement="bottomLeft"
      transitionName="slide-up"
      value={"2019-10-29T20:00:00.000Z"}
    />
  </Col>
  <Col
    span={6}
  >
    <Styled(ButtonGroup)>
      <Button
        block={false}
        disabled={true}
        ghost={false}
        htmlType="button"
        icon="minus"
        loading={false}
      />
      <Button
        block={false}
        disabled={true}
        ghost={false}
        htmlType="button"
        icon="plus"
        loading={false}
      />
    </Styled(ButtonGroup)>
  </Col>
</Row>
`;
